### ISSUE TITLE
Lighting flickering on Exynos in AC:NL, TLoZ:ALTTP

### ISSUE THREAD

**Issue by nergzd723** [citra-emu/citra-android#11]
Lighting flickers every second frame or so on Exynos 9810.(is probably an issue with Mali GPU)

I don't know where logs are saved in mobile version, and there's no citra_log.txt in my /sdcard.

The issue is visible on attached screenshots.
![Screenshot_20200523-092348_Citra](https://user-images.githubusercontent.com/53225591/82723384-2b810980-9cd7-11ea-8b57-de557a8e26d0.jpg)
![Screenshot_20200523-092342_Citra](https://user-images.githubusercontent.com/53225591/82723389-2f149080-9cd7-11ea-975d-5ae357f38784.jpg)

**Comment by Developer-exe** [comment]
You can find the log file on the device's internal storage here:
/storage/sdcard0/citra-emu/log/citra_log.txt

**Comment by nergzd723** [comment]
May be a shader issue, since water actually changes in ALBW. Screenshots and log attached.

![Screenshot_20200524-100937_Citra](https://user-images.githubusercontent.com/53225591/82747955-b2e97e00-9da6-11ea-8c40-727e41738e38.jpg)
![Screenshot_20200524-100931_Citra](https://user-images.githubusercontent.com/53225591/82747956-b5e46e80-9da6-11ea-8a1e-5c790453f084.jpg)


[citra_log.txt](https://github.com/citra-emu/citra-android/files/4673074/citra_log.txt)

**Comment by Dragios** [comment]
### Lighting flickering in all (most) games
I think it happens on all games (most if not all). Maybe only affect with _Mali GPU_ users. Does it occur on Adreno GPU?
Some games flicker so fast that might cause for potential seizures.
Phone tested: *Samsung S21 Ultra 5G (Exynos variant)*

| Game | Frame 1 | Frame 2 |
| - | - | - |
| Pokémon X| ![Screenshot_20210209-175737_Citra](https://user-images.githubusercontent.com/18496868/107347945-58917880-6b01-11eb-83bf-58c10f829c1d.png) | ![Screenshot_20210209-175746_Citra](https://user-images.githubusercontent.com/18496868/107347951-5a5b3c00-6b01-11eb-89dc-2f4572739c11.png) | 
| Phoenix Wright: Ace Attorney - Dual Destinies | ![Screenshot_20210209-175911_Citra](https://user-images.githubusercontent.com/18496868/107348231-b7ef8880-6b01-11eb-834f-b4eb85554d52.png) | ![Screenshot_20210209-175823_Citra](https://user-images.githubusercontent.com/18496868/107348241-bb830f80-6b01-11eb-848a-17e8ae7bd336.png) |
| The Great Ace Attorney | ![Screenshot_20210209-180020_Citra](https://user-images.githubusercontent.com/18496868/107348253-bf169680-6b01-11eb-8472-7654d1432223.png) | ![Screenshot_20210209-180012_Citra](https://user-images.githubusercontent.com/18496868/107348260-c047c380-6b01-11eb-8aa5-32536cd8a491.png) |

**Comment by nergzd723** [comment]
> ### Lighting flickering in all (most) games
> I think it happens on all games (most if not all). Maybe only affect with _Mali GPU_ users. Does it occur on Adreno GPU?
> Some games flicker so fast that might cause for potential seizures.
> Phone tested: _Samsung S21 Ultra 5G (Exynos variant)_
> 
> Game	Frame 1	Frame 2
> Pokémon X	![Screenshot_20210209-175737_Citra](https://user-images.githubusercontent.com/18496868/107347945-58917880-6b01-11eb-83bf-58c10f829c1d.png)	![Screenshot_20210209-175746_Citra](https://user-images.githubusercontent.com/18496868/107347951-5a5b3c00-6b01-11eb-89dc-2f4572739c11.png)
> Phoenix Wright: Ace Attorney - Dual Destinies	![Screenshot_20210209-175911_Citra](https://user-images.githubusercontent.com/18496868/107348231-b7ef8880-6b01-11eb-834f-b4eb85554d52.png)	![Screenshot_20210209-175823_Citra](https://user-images.githubusercontent.com/18496868/107348241-bb830f80-6b01-11eb-848a-17e8ae7bd336.png)
> The Great Ace Attorney	![Screenshot_20210209-180020_Citra](https://user-images.githubusercontent.com/18496868/107348253-bf169680-6b01-11eb-8472-7654d1432223.png)	![Screenshot_20210209-180012_Citra](https://user-images.githubusercontent.com/18496868/107348260-c047c380-6b01-11eb-8aa5-32536cd8a491.png)

No, it does not happen on adreno. Anyway, weihuoya pushed a fix which fixes the flickering issue and the performance on mali, see https://github.com/weihuoya/citra/commit/5aab825724b4c93b7381df2691f02abdbec4196e . I hope it gets merged here soon. (I actually get acceptable performance in dgs and aa5 with it on exynos 9810 :D)

**Comment by fokkonaut** [comment]
May I ask why that commit did not get merged yet? I have a S9 and would want to try if that fixes my issues.
